src/libraries/Microsoft.PowerFx.Interpreter/Environment/BasicServiceProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.PowerFx
+{    
+    /// <summary>
+    /// Trivial <see cref="IServiceProvider"/> implementation that allows chaining and composition. 
+    /// </summary>
+    public sealed class BasicServiceProvider : IServiceProvider
```

Comment:
Previously, the IServiceProvider implementation was on the ReadOnlySymbolValues. Now it's split out to a separate class.